src/Grpc.Core.Api/AsyncClientStreamingCall.cs

```diff
@@ -164,4 +168,23 @@ public void Dispose()
     {
         callState.Dispose();
     }
+
+    private string DebuggerToString() => CallDebuggerHelpers.DebuggerToString(callState);
+
+    private sealed class AsyncClientStreamingCallDebugView
+    {
+        private readonly AsyncClientStreamingCall<TRequest, TResponse> _call;
+
+        public AsyncClientStreamingCallDebugView(AsyncClientStreamingCall<TRequest, TResponse> call)
+        {
+            _call = call;
+        }
+
+        public bool IsComplete => CallDebuggerHelpers.GetStatus(_call.callState) != null;
+        public Status? Status => CallDebuggerHelpers.GetStatus(_call.callState);
+        public Metadata? ResponseHeaders => _call.ResponseHeadersAsync.Status == TaskStatus.RanToCompletion ? _call.ResponseHeadersAsync.GetAwaiter().GetResult() : null;
```

Comment:
Would it be better to report null in the debug view rather than potentially hang the debugger with a sync-over-async to out-of-proc data to a call that might be broken?